javalib/src/main/scala/java/util/concurrent/CopyOnWriteArrayList.scala

```diff
@@ -274,43 +288,46 @@ class CopyOnWriteArrayList[E <: AnyRef] private (private var inner: js.Array[E])
   }
 
   protected def innerGet(index: Int): E =
-    inner(index)
+    innerImpl.get(inner, index).asInstanceOf[E]
 
   protected def innerSet(index: Int, elem: E): Unit =
-    inner(index) = elem
+    innerImpl.set(inner, index, elem)
 
-  protected def innerPush(elem: E): Unit =
-    inner.push(elem)
+  protected def innerPush(elem: E): Unit = {
+    ensureCapacity(size() + 1)
```

Comment:
In `CopyOnWriteArrayList`, we should keep the regular notion of capacity (without the `+ 1`). It made sense in `PriorityQueue` because the generic implementation avoids index 0 anyway. But in `CopyOnWriteArrayList`, avoiding 0 is a Wasm-only thing, due to storing the length.

We should concentrate all those `+ 1`s and `- 1`s inside `JArrayImpl`.